phones/tests/mgmt_delete_phone_data_tests.py

```diff
@@ -0,0 +1,272 @@
+from datetime import datetime, timezone, timedelta
+from io import StringIO
+from typing import Iterator
+from unittest.mock import Mock, patch
+
+from django.contrib.auth.models import User
+from django.core.management import call_command, CommandError
+from django.conf import settings
+
+from allauth.socialaccount.models import SocialAccount
+from model_bakery import baker
+from pytest_django.fixtures import SettingsWrapper
+import pytest
+
+if settings.PHONES_ENABLED:
+    from ..models import InboundContact, RealPhone, RelayNumber
+
+pytestmark = pytest.mark.skipif(
+    not settings.PHONES_ENABLED, reason="PHONES_ENABLED is False"
+)
+THE_COMMAND = "delete_phone_data"
+
+
+@pytest.fixture(autouse=True)
+def test_settings(settings: SettingsWrapper) -> SettingsWrapper:
+    """Override settings for tests"""
+    settings.SUBSCRIPTIONS_WITH_PHONE = ["PHONE_SUBSCRIPTION"]
+    return settings
+
+
+@pytest.fixture(autouse=True)
+def mock_twilio_client() -> Iterator[Mock]:
+    """Mock PhonesConfig with a mock twilio client"""
+    with patch("phones.apps.PhonesConfig.twilio_client") as mock_twilio_client:
+        yield mock_twilio_client
```

Comment:
question (non-blocking): interesting. does this management command trigger a Twilio API call?

Reply:
No, it's creating the test data. Creating a `RealPhone` calls [realphone_post_save](https://github.com/mozilla/fx-private-relay/blob/62db400cdd4ad1a57627d8d06fe2536ba0bfcc74/phones/models.py#L192), which calls Twilio to send a text message, and creating a `RelayNumber` will call [.save](https://github.com/mozilla/fx-private-relay/blob/62db400cdd4ad1a57627d8d06fe2536ba0bfcc74/phones/models.py#L263) which registers the number with Twilio and adds US numbers to a message pool.

I think mocking will only work in the Twilio case, not the Intelliquint case, so I think I need to fix that stuff first.

Reply:
Oh no worries. Just curios.

Reply:
I'm adding `settings.PHONES_NO_CLIENT_CALLS_IN_TEST`, and using it to skip calling Twilio or IQ clients for code like this that is not testing the client interaction. I've got bigger ideas for later.